src/fasta.rs

```diff
@@ -1,209 +1,932 @@
+            s,
+            FastaParser::<String>::new(FastaParseSettings {
+                concatenate_headers,
+                allow_preceding_comment: true
+            }),
+            expected
+        );
+        assert_parse!(
+            s,
+            FastaParser::<String>::new(FastaParseSettings {
+                concatenate_headers,
+                allow_preceding_comment: false
+            }),
+            expected
+        );
+    }
+
+    /// Assert a parse matches the expected value with the given value of allow_preceding_comment,
+    /// and all values for other settings
+    fn assert_parse_with_allow_preceding_comment(
+        s: &str,
+        allow_preceding_comment: bool,
+        expected: Vec<FastaRecord<String>>,
+    ) {
+        assert_parse!(
+            s,
+            FastaParser::<String>::new(FastaParseSettings {
+                concatenate_headers: true,
+                allow_preceding_comment
+            }),
+            expected
+        );
+        assert_parse!(
+            s,
+            FastaParser::<String>::new(FastaParseSettings {
+                concatenate_headers: false,
+                allow_preceding_comment
+            }),
+            expected
+        );
+    }
+
     #[test]
     fn test_empty_fasta() {
-        let r = SimpleFastaParser::parse("".as_bytes()).unwrap();
-        assert_eq!(r.len(), 0);
+        assert_parse_with_all_settings("", vec![]);
     }
 
     #[test]
     fn test_empty_fasta_with_newlines() {
-        let r = SimpleFastaParser::parse("\n\n".as_bytes()).unwrap();
-        assert_eq!(r.len(), 0);
+        assert_parse_with_all_settings("\n\n", vec![]);
     }
 
     #[test]
     fn test_empty_fasta_with_many_newlines() {
-        let r = SimpleFastaParser::parse("  \n\n \n  \r\r \r\n  \r \n".as_bytes()).unwrap();
-        assert_eq!(r.len(), 0);
+        assert_parse_with_all_settings("  \n\n \n  \r\r \r\n  \r \n", vec![]);
     }
 
     #[test]
     fn test_fasta_with_no_content() {
-        let r = SimpleFastaParser::parse(">Virus\n".as_bytes()).unwrap();
-        assert_eq!(r.len(), 1);
-        assert_eq!(r[0], ("Virus".to_string(), "".to_string()));
+        assert_parse_with_all_settings(
+            ">Virus\n",
+            vec![FastaRecord {
+                header: "Virus".to_string(),
+                contents: "".to_string(),
+                line_range: (1, 2),
+            }],
+        );
     }
 
     #[test]
     fn test_fasta_with_empty_content() {
-        let r = SimpleFastaParser::parse(">Virus\n\n".as_bytes()).unwrap();
-        assert_eq!(r, vec![("Virus".to_string(), "".to_string())]);
+        assert_parse_with_all_settings(
+            ">Virus\n\n",
+            vec![FastaRecord {
+                header: "Virus".to_string(),
+                contents: "".to_string(),
+                line_range: (1, 3),
+            }],
+        );
+    }
+
+    #[test]
+    fn test_comment_fasta_with_allow_preceding_comment() {
```

Comment:
what happens if the setting is the default `false`? I only see happy case tests below